flytekit/remote/remote.py

```diff
@@ -1028,11 +1028,14 @@ def _get_image_names(entity: typing.Union[PythonAutoContainerTask, WorkflowBase]
             if isinstance(entity, WorkflowBase):
                 default_inputs = entity.python_interface.default_inputs_as_kwargs
 
+            from flytekit.configuration.plugin import get_plugin
+            additional_context = get_plugin().get_additional_context(entity)
+
             # The md5 version that we send to S3/GCS has to match the file contents exactly,
             # but we don't have to use it when registering with the Flyte backend.
             # For that add the hash of the compilation settings to hash of file
             version = self._version_from_hash(
-                md5_bytes, serialization_settings, default_inputs, *_get_image_names(entity)
+                md5_bytes, serialization_settings, default_inputs, *_get_image_names(entity), *additional_context
```

Comment:
My feeling is that core logic i.e. the default version hash logic, should remain in its section, rather than moved to the plugin methods. This mostly follows the existing pattern I noticed in FlytekitPlugin, where core logic is not often moved to its methods – however there are some exceptions, as I suppose with `FlytekitPlugin.get_remote`